server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -151,19 +155,22 @@ private static BulkItemResultHolder executeDeleteRequest(final DeleteRequest del
                                                              final IndexShard primary,
                                                              final MappingUpdatePerformer mappingUpdater) throws Exception {
         Engine.DeleteResult deleteResult = executeDeleteRequestOnPrimary(deleteRequest, primary, mappingUpdater);
-        if (deleteResult.hasFailure()) {
-            return new BulkItemResultHolder(null, deleteResult, bulkItemRequest);
-        } else {
-            DeleteResponse response = new DeleteResponse(primary.shardId(), deleteRequest.type(), deleteRequest.id(),
+        switch (deleteResult.getResultType()) {
+            case SUCCESS:
+                DeleteResponse response = new DeleteResponse(primary.shardId(), deleteRequest.type(), deleteRequest.id(),
                     deleteResult.getSeqNo(), primary.getPrimaryTerm(), deleteResult.getVersion(), deleteResult.isFound());
-            return new BulkItemResultHolder(response, deleteResult, bulkItemRequest);
+                return new BulkItemResultHolder(response, deleteResult, bulkItemRequest);
+            case FAILURE:
+                return new BulkItemResultHolder(null, deleteResult, bulkItemRequest);
```

Comment:
I think you should have all options listed here and don't use default. Be explicit here please

Reply:
sure. I'll add en explicit line + a default for future additions in the enum